apache2/modsecurity.c

```diff
@@ -266,7 +275,10 @@ void modsecurity_child_init(msc_engine *msce) {
 
     set_lock_args(lock_args, WAFJSONLOG_LOCK_ID);
 
-    waf_create_lock(msce->wafjsonlog_lock, lock_args);    
+    waf_create_lock(msce->wafjsonlog_lock, lock_args); 
+
+    signal(SIGUSR1, modsecurity_handle_signals_for_reopen); 
```

Comment:
Let's use `sigaction()` instead of `signal()` for the latter is deprecated. See more on the subject here: https://stackoverflow.com/questions/231912/what-is-the-difference-between-sigaction-and-signal

Reply:
changed